### PR TITLE
feat: to function writes to remote influxdb

### LINKED ISSUE #1259: Using to function with remote influxdb instance fails
I was trying to write my OSS metrics into my cloud instance. When entering in the remote bucket, org, url, and token into the To function, i am unable to write data. My suspicion is that the query needs to use the token in the to function instead of the token that is used to run the query. 

I used an actual token generated from the cloud instance, not the word `TOKEN`.

```
from(bucket: "telegraf")
  |> range(start: -15m, stop: now())
  |> to(bucket: "oss_metrics", 
        org: "russ+cloud2's organization", 
        host: "https://us-west-2-1.aws.cloud2.influxdata.com", 
        token: "TOKEN")
```

image

### PR DESCRIPTION
**Pull request by lesam**
Closes #1259

### Done checklist
- [ ] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: dependencies/influxdb/http.go
Comment (jsternberg) [inline dependencies/influxdb/http.go:7]: I didn't know there was a v2 now. Looks good.

File: dependencies/influxdb/http.go
Comment (jsternberg) [inline dependencies/influxdb/http.go:76]: Can you make this a global function so it can be called in other methods potentially?

The primary reason this package doesn't use the influxdb go client is because there was a v0 version being used in one project that depends on flux and there was a newer v1 and they had different APIs so we weren't able to use the official API. This package should be using the official API everywhere if it's available and I have a feeling this functionality will be needed for the query methods too.

File: dependencies/influxdb/provider.go
Comment (jsternberg) [inline dependencies/influxdb/provider.go:110]: Can you rename this to `Writer` so it has parity with `Reader`?

File: dependencies/influxdb/provider.go
Comment (jsternberg) [inline dependencies/influxdb/provider.go:112]: Use `io.Closer` here.

File: dependencies/influxdb/service.go
Comment (jsternberg) [inline dependencies/influxdb/service.go:1]: Can you either put this file into an internal package within this directory or avoid exporting anything in this file? The packages in this directory are particularly susceptible to API changes because they are intended to be implemented by third-parties that are embedding flux. I'd like to avoid creating a contract for a fork of a dependency.

File: dependencies/influxdb/http.go
Comment (jsternberg) [inline dependencies/influxdb/http.go:472]: This should just be:

```
var _ PointsWriter = httpWriter{}
```

File: stdlib/influxdata/influxdb/to.go
Comment (jsternberg) [inline stdlib/influxdata/influxdb/to.go:4]: Sort the imports.

File: stdlib/influxdata/influxdb/to.go
Comment (jsternberg) [inline stdlib/influxdata/influxdb/to.go:30]: You can remove this line. It doesn't do anything but remains in the codebase in a lot of places because it doesn't cause any harm and nobody has gone back to remove them.

File: stdlib/influxdata/influxdb/to.go
Comment (jsternberg) [inline stdlib/influxdata/influxdb/to.go:65]: I think this might be unused now that it is in `dependencies/influxdb`.

File: stdlib/influxdata/influxdb/to.go
Comment (jsternberg) [inline stdlib/influxdata/influxdb/to.go:70]: Same here. I don't think this is needed.

File: stdlib/influxdata/influxdb/to.go
Comment (jsternberg) [inline stdlib/influxdata/influxdb/to.go:599]: If you can make these unexported that would be good.

File: stdlib/influxdata/influxdb/to.go
Comment (jsternberg) [inline stdlib/influxdata/influxdb/to.go:622]: Newline at the end here.

**Review by jsternberg — CHANGES_REQUESTED** [review]
Looks good. A few style changes and suggestions. Make sure you `go fmt` everything and sort the imports since I saw a few files where that wasn't the case.

Do you just need this to work for the flux library itself or do you also need it to work in influxdb/idpe? This change will work fine to get the flux CLI to work, but we'll need some follow up work to integrate these same changes into influxdb and idpe. I don't think these changes should break the functionality of either of those.

File: dependencies/influxdb/service.go
Comment (lesam) [reply dependencies/influxdb/service.go:1]: I plan to remove this file before merge - the only difference from the copy is that it takes a `Doer` interface instead of a `http.Client` struct. I think this can be changed upstream before merge.

**Comment by lesam** [comment]
I only need this change for the flux library itself - I intend to pick it up in Kapacitor where we don't overwrite `to`.

I'm not sure how this affects the Kafka `to` that's also in this repo - is there a test I can run for that?

Also, what tests should be added with this PR?

File: dependencies/influxdb/service.go
Comment (lesam) [reply dependencies/influxdb/service.go:1]: Specifically I need this PR merged upstream: https://github.com/influxdata/influxdb-client-go/pull/256/files

Comment (lesam) [reply dependencies/influxdb/service.go]: Actually let's not block this on the other PR - I put in a TODO referencing the upstream PR and moved the service implementation to `internal`

File: dependencies/influxdb/http.go
Comment (lesam) [inline dependencies/influxdb/http.go:473]: The error handling here is a bit ugly, there is an open issue for the client to improve this with a v3 of the client.

File: stdlib/influxdata/influxdb/to.go
Comment (lesam) [inline stdlib/influxdata/influxdb/to.go:35]: This is mostly stolen from idpe, though idpe uses v0 of `influxdb-client-go`

File: dependencies/influxdb/http.go
Comment (jsternberg) [inline dependencies/influxdb/http.go:467]: Leave this unexported too. I don't want to create an external api that others can rely on.

File: dependencies/influxdb/provider.go
Comment (jsternberg) [inline dependencies/influxdb/provider.go:110]: Godoc comment wasn't updated. Also end it with a period.

File: dependencies/influxdb/provider.go
Comment (jsternberg) [inline dependencies/influxdb/provider.go:153]: Rename to `IdOrName()` and document what this does.

File: stdlib/influxdata/influxdb/to.go
Comment (jsternberg) [inline stdlib/influxdata/influxdb/to.go:546]: I don't think that url or brokers are valid arguments for this to function. I assume this was probably when `to()` was going to be an all-encompassing function that included choosing the correct implementation automatically. Originally, from was going to take multiple arguments and choose which underlying from to call depending on those arguments.

With that, I think you can remove the entirety of the kafka related stuff here and either do this:

```go
s := &ToOpSpec{}
if err := s.ReadArgs(args); err != nil {
    return nil, err
}
return s, nil
```

Or you can remove that `ReadArgs` function altogether and replace the implementation here with it.

File: dependencies/influxdb/internal/service.go
Comment (jsternberg) [inline dependencies/influxdb/internal/service.go:15]: Can you rename this to something like `influxhttp`, `ihttp`, `httpapi`, or `apihttp`? I'd like a name other than `http2` that is a bit more descriptive.

**Review by jsternberg — CHANGES_REQUESTED** [review]
I think we're almost there. A few more changes. I didn't catch the kafka stuff the first time but the kafka stuff can be removed. It's not used and it's not even possible to reach that code.

File: stdlib/influxdata/influxdb/to.go
Comment (lesam) [reply stdlib/influxdata/influxdb/to.go]: I kept ReadArgs - I think it reads a bit more nicely.

**Review by jsternberg — APPROVED** [review]
I think this looks good. The only condition is to please compile this into influxdb and make sure that to still works in it. If everything is correct, the http `to()` won't work in influxdb but the normal `to()` will still work. We'll get them all integrated together but this is the first step.

File: dependencies/influxdb/http.go
Comment (lesam) [inline dependencies/influxdb/http.go:502]: Note this 'TrimRight' to avoid double-writing `\n`

File: dependencies/influxdb/http_test.go
Comment (lesam) [inline dependencies/influxdb/http_test.go:1]: This is a newly written test for `httpWriter`

**Comment by lesam** [comment]
> I think this looks good. The only condition is to please compile this into influxdb and make sure that to still works in it. If everything is correct, the http `to()` won't work in influxdb but the normal `to()` will still work. We'll get them all integrated together but this is the first step.

There will need to be some influxdb and IDPE changes when this merges - mainly we can't use the same name for `ToOpSpec` in both places. 1.x does this more correctly by using its own name, we will have to update 2.x and likely IDPE similarly.

File: stdlib/influxdata/influxdb/to_test.go
Comment (lesam) [inline stdlib/influxdata/influxdb/to_test.go:1]: Stolen from IDPE

File: mock/provider.go
Comment (jsternberg) [inline mock/provider.go:9]: Can you rename this and the file `InfluxDBProvider` and `influxdb_provider.go` respectively?

**Review by jsternberg — APPROVED** [review]
One change to a name but pre-approving since the rest is good. If you make that one change, feel free to merge.

**Pull request merged**
